paddle/fluid/distributed/fleet_executor/carrier.cc

```diff
@@ -80,12 +83,23 @@ void Carrier::Release() {
 
 Carrier::~Carrier() { VLOG(3) << "Carrier's destructor."; }
 
-void Carrier::CopyParameters(int microbatch_id,
-                             const framework::ProgramDesc& program) {
+void Carrier::CopyParameters(
+    int microbatch_id, const framework::ProgramDesc& program,
+    const std::vector<std::string>& force_root_scope_var_names) {
   auto& global_block = program.Block(0);
 
+  std::map<std::string, int> force_root_tmp_map;
+  for (auto var_name : force_root_scope_var_names) {
+    force_root_tmp_map.insert({var_name, 1});
+  }
   for (auto& var : global_block.AllVars()) {
-    if (var->Persistable() && microbatch_id == 0) {
+    std::string var_name = var->Name();
+    bool force_root =
+        force_root_tmp_map.find(var_name) != force_root_tmp_map.end();
+    if (force_root) {
+      VLOG(3) << var_name << " will be forced to be created in the root scope.";
```

Comment:
If there were many vars, this line will output redundancy logs.
So use VLOG 4+ level?

Reply:
done updated